setup.sh

```diff
@@ -1,45 +1,109 @@
 #!/bin/bash
 
+#  Copyright 2023 DATA @ UHN. See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+# Check the user's authorship configuration
+if ! $(git config user.name 1> /dev/null) || ! $(git config user.email 1> /dev/null)
+then
+  whiptail --backtitle "New CARDS repository setup" --msgbox "You must configure the user name and email address with \"git config user.name Some Name\" and \"git config user.email somebody@email.com\" and then rerun this script. Exiting." 9 78
+  exit 1
+fi
+
+if ! whiptail --backtitle "New CARDS repository setup" --title "Please confirm git authorship details" --yesno "Committing as $(git config user.name) <$(git config user.email)>, OK?" 8 78
+then
+  whiptail --backtitle "New CARDS repository setup" --msgbox "You can configure the user name and email address with \"git config user.name Some Name\" and \"git config user.email somebody@email.com\" and then rerun this script. Exiting." 9 78
+  exit 1
+fi
+
 # Ask about the generic CARDS version to use
 
-# Get the latest CARDS version from Nexus
-CARDS_VERSION=$(curl -s https://nexus.phenotips.org/nexus/content/repositories/releases/io/uhndata/cards/cards-modules/maven-metadata.xml | grep --max-count=1 '<release>' | cut '-d>' -f2 | cut '-d<' -f1)
-if [[ -z $CARDS_VERSION ]]
+# Get the available CARDS versions from Maven Central
+declare -a RELEASED_VERSIONS=( $(curl -s https://repo1.maven.org/maven2/io/uhndata/cards/cards-modules/maven-metadata.xml | grep '<version>' | tac | cut '-d>' -f2 | cut '-d<' -f1) )
+declare -i RELEASED_VERSIONS_COUNT=${#RELEASED_VERSIONS[*]}
+# Look on the local disk
+LOCAL_CARDS_VERSION=$(grep --max-count=1 '^  <version>' ../cards/pom.xml 2>/dev/null | cut '-d>' -f2 | cut '-d<' -f1)
+declare VERSION
+declare HAS_DEFAULT
+for (( i=0 ; i<RELEASED_VERSIONS_COUNT; i++ ))
+do
+  VERSION="${VERSION} ${RELEASED_VERSIONS[$i]} ${RELEASED_VERSIONS[$i]} ${HAS_DEFAULT:-ON} "
+  HAS_DEFAULT=OFF
+done
+if [[ -n $LOCAL_CARDS_VERSION ]]
+then
+  VERSION="${LOCAL_CARDS_VERSION} ${LOCAL_CARDS_VERSION} ${HAS_DEFAULT:-ON} ${VERSION}"
+  HAS_DEFAULT=OFF
+fi
```

Comment:
I think this should be amended so that `$LOCAL_CARDS_VERSION` is not listed if it is already included in the `$RELEASED_VERSIONS` list. For example, if the local version of CARDS is `0.9.22`, `0.9.22` would be listed twice as it is available both locally and in the list of released versions obtained from the Maven repository.

But given my above comment regarding switching from checking `../cards/pom.xml` for the locally available CARDS version to looking under `~/.m2/repository/io/uhndata/cards/cards` we could end up having more than one locally available version. Therefore, what would be best would be to list all locally available versions as well as the versions listed on the Maven repository but ensuring that no single version gets listed more than once.